### ISSUE TITLE
Lack of helpful error message when crashing due to global file OUTVAR option error

### ISSUE THREAD

**Issue by orianac** [UW-Hydro/VIC#209]
VIC4.2.b crashed when using a faulty global file, but exited with a vague memory error instead of a helpful error message.

The problematic part of my global file looked like this:

```
OUTFILE         fluxes  4
OUTVAR          OUT_PREC
OUTVAR          OUT_EVAP
OUTVAR          OUT_RUNOFF
OUTVAR          OUT_BASEFLOW
OUTVAR          OUT_AIR_TEMP
OUTVAR          OUT_SOIL_LIQ
OUTVAR          OUT_SWE
```
(The 4 should have been a 7. Upon fixing the numeric typo the model ran successfully to completion.)

The model did not crash but rather hang in limbo after it read the soil file. The error I got was this:
```
*** glibc detected *** /home/oriana/scripts/vic/VIC_jhamman/src/vicNl: free(): invalid size: 0x0000000000e52d00 *** 
```

**Comment by jhamman** [comment]
Thanks @orianac.  Indeed, the error given is not helpful in this situation.  We should either:
a) calculate the number of output variables on-the-fly, eliminating the need for the `4` in your example, or
b) enforce that they match.

It seems easier to enforce that they match (in terms of writing the code).  However, I think it is better form to reduced the redundant information of the number of variables.

I'm labeling this issue as both documentation and an enhancement for VIC.5.

Lastly, a bit more information on the error.  When run through a debugger, the memory error was found to occur in `set_output_var()` in `output_list_utils.c`.

**Comment by bartnijssen** [comment]
Agreed on reducing the redundant info if we can make that happen easily. It should be labeled as a bug as well, since you should not be able to make the model crash by just mis-specifying a number in the input. I'll assign it to me to come up with a hotfix for 4.2.

**Comment by jhamman** [comment]
closed via #364
